alacritty/src/event.rs

```diff
@@ -74,17 +81,49 @@ impl From<TerminalEvent> for Event {
     }
 }
 
+/// Regex search state.
+pub struct SearchState {
+    /// Search string regex.
+    regex: Option<String>,
+
+    /// Search direction.
+    direction: Direction,
+
+    /// Change in display offset since the beginning of the search.
+    display_offset_delta: isize,
+
+    /// Vi cursor position before search.
+    vi_cursor_point: Point,
+}
+
+impl SearchState {
+    fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Default for SearchState {
+    fn default() -> Self {
+        Self {
+            direction: Direction::Right,
+            display_offset_delta: 0,
+            vi_cursor_point: Point::default(),
+            regex: None,
+        }
+    }
+}
```

Comment:
I kind of feel like those things shouldn't be in `event.rs`, as well as some other search related stuff, the file is already kind of huge.

Reply:
Where else would you put it? It's event state.

Reply:
it's not really an event state, it's a search state. it's even called `SearchState`, not `SearchEventState`, and event is something we're getting or sending, not something we persist on `ActionContext`. Though we already have `DisplayUpdate`, etc in that particular. Maybe we should keep it here then.

Reply:
It's an event state, just like `Mouse` is. It's a state solely maintained for the event processor.